sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -571,12 +574,13 @@ def assign_rows_csr(
     out : array, shape=(arbitrary, n_features)
     """
     cdef:
-        # npy_intp (np.intp in Python) is what np.where returns,
+        # intptr_t (npy_intp, np.intp in Python) is what np.where returns,
         # but int is what scipy.sparse uses.
         int i, ind, j, k
-        cnp.npy_intp rX
+        intptr_t rX
         const floating[:] data = X.data
-        const int[:] indices = X.indices, indptr = X.indptr
+        const int[:] indices = X.indices
+        const int[:] indptr = X.indptr
```

Comment:
I think scipy most uses `int32_t` and `int64_t` as the need arises in recent version of scipy, no?

Reply:
`int` and `int32_t` are equivalent. SciPy supports both `int32_t` and `int64_t`. This implementation could be ported to support both in another PR.

Reply:
I'll change at least to int32_t to be more explicit here. But I don't want this PR to add int64 support here.